### PR TITLE
[GLASS] Get questionnaire datasets from datastore

### PR DESCRIPTION
**Pull request by deeonwuli**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/86959a3ky

### :memo: Implementation
- [x] Get module questionnaires from dataStore
- [x] Show questionnaire table column based on module

### :video_camera: Screenshots/Screen capture

https://github.com/user-attachments/assets/c3897859-8a97-4bc8-8ed1-4d2fb6d205cd

### :fire: Notes to the tester
```
REACT_APP_DHIS2_BASE_URL=https://dev.eyeseetea.com/who-dev-238/
REACT_APP_REPORT_VARIANT=glass-submission
```

#86959a3ky

### CODE REVIEW

File: src/data/reports/glass-data-submission/GLASSDataSubmissionDefaultRepository.ts [hunk 1]
```diff
@@ -147,7 +149,13 @@ export class GLASSDataSubmissionDefaultRepository implements GLASSDataSubmission
             enrolledCountries.join(";"),
             modules
         );
-        const registrations = await this.getRegistrations(dataSubmissionItems);
+
+        const questionnaireDataSets =
```
Comment (9sneha-n) [inline src/data/reports/glass-data-submission/GLASSDataSubmissionDefaultRepository.ts:153]: can you rename questionnaireDataSets to questionnaires as modules may have questionnaires that are program as well.

File: src/webapp/reports/glass-data-submission/glass-data-submission-list/DataSubmissionList.tsx [hunk 2]
```diff
@@ -134,9 +144,9 @@ export const DataSubmissionList: React.FC = React.memo(() => {
                     multiple: true,
                     onClick: async (selectedIds: string[]) =>
                         updateDataSubmissionStatus("accept", selectedIds, Namespaces.DATA_SUBMISSSIONS),
-                    isActive: (rows: DataSubmissionViewModel[]) => {
-                        return _.every(rows, row => row.status === "PENDING_UPDATE_APPROVAL");
-                    },
+                    // isActive: (rows: DataSubmissionViewModel[]) => {
```
Comment (9sneha-n) [inline src/webapp/reports/glass-data-submission/glass-data-submission-list/DataSubmissionList.tsx:147]: why have we commented this code?

File: src/webapp/reports/glass-data-submission/glass-data-submission-list/useDataSubmissionList.tsx [hunk 3]
```diff
@@ -27,12 +29,21 @@ export function useDataSubmissionList(filters: Filter) {
         filters
     );
 
+    const questionnaireDataSets = useMemo(() => {
```
Comment (9sneha-n) [inline src/webapp/reports/glass-data-submission/glass-data-submission-list/useDataSubmissionList.tsx:32]: this is not being used anywhere?

**Review by 9sneha-n — COMMENTED** [review]
@deeonwuli Looks good overall, have a few minor comments.

File: src/webapp/reports/glass-data-submission/glass-data-submission-list/useDataSubmissionList.tsx (hunk 3 above)
Comment (deeonwuli) [reply src/webapp/reports/glass-data-submission/glass-data-submission-list/useDataSubmissionList.tsx:32]: i'm using this to filter for which columns to show in the table. if there are no questionnaires for the selected module, the Questionnaire Completed column is not included in the table

File: src/webapp/reports/glass-data-submission/glass-data-submission-list/DataSubmissionList.tsx (hunk 2 above)
Comment (deeonwuli) [reply src/webapp/reports/glass-data-submission/glass-data-submission-list/DataSubmissionList.tsx:147]: oversight during testing 😅

File: src/webapp/reports/glass-data-submission/glass-data-submission-list/useDataSubmissionList.tsx (hunk 3 above)
Comment (9sneha-n) [reply src/webapp/reports/glass-data-submission/glass-data-submission-list/useDataSubmissionList.tsx]: ah yes, i see it now.

**Review by 9sneha-n — APPROVED** [review]
@deeonwuli Thank you

**Review by bhavananarayanan — APPROVED** [review]
Approved, thank you

**Pull request merged**